src/Stores/Promises/selectors.js

```diff
@@ -23,9 +23,22 @@ export const promiseSubjectsSelector: (
       const links = subjectsPromises
         ? subjectsPromises.filter(sP => sP.promise === promise.id)
         : [];
-      const subs = links.map(
-        link => (subjects ? subjects.find(s => link.subject === s.id) : [])
-      );
+      const subs = links.map(link => {
+        if (!subjects) {
+          // TODO trigger loading of subjects instead of throwing error
+          throw new Error('trying to find subjects when they are not loaded');
+        }
+        const subject = {
+          ...subjects.find(s => link.subject === s.id),
```

Comment:
possible null pointer exception if the subject is not found in the array? Cant spread null / undefined (is that ever going to happen?)

Reply:
Added a check for this.

It shouldn't happen. But http://pragmatictips.com/33 says we should **assert** it does not happen